hbase_storage/src/main/java/com/splicemachine/si/data/hbase/coprocessor/SIObserver.java

```diff
@@ -193,6 +172,29 @@ public void postCompact(ObserverContext<RegionCoprocessorEnvironment> c, Store s
         }
     }
 
+    private PurgeConfig getPurgeConfig(final FlushLifeCycleTracker tracker, final SConfiguration conf) throws IOException {
+        PurgeConfig purgeConfig;
+        if(tracker instanceof FlushLifeCycleTrackerWithConfig) {
+            purgeConfig = ((FlushLifeCycleTrackerWithConfig) tracker).getConfig();
+        } else {
+            SpliceLogUtils.warn(LOG, "Splice store flusher is not set, as a result flush will ignore minimum retention period " +
```

Comment:
I think we should be conservative here and when this is not set, we shouldn't do any purging during flush.

Reply:
That's definitely much better. I will still issue a warning that purge during flush will not take place and the configuration should be fixed.
Thanks!